### ISSUE TITLE
Deprecate lib.bzl

### ISSUE THREAD

**Issue by Capstan** [bazelbuild/bazel-skylib#82]
This process was started in commit 1099dd2 by @thomasvl .

1. Use a `print()` warning.
1. Wait one or two releases. ← we are here.
1. Use a `fail()` error.
1. Remove `lib.bzl`

**Comment by manekinekko** [comment]
Hi, I got a quick question about this deprecation. If I got this right:

BEFORE:  `load("@bazel_skylib//:lib.bzl", "paths")`
AFTER:  `load("@bazel_skylib//lib:paths.bzl", "paths")`

Right?

**Comment by Capstan** [comment]
Correct. Or if you want to be pedantic about not re-exporting:

`load("@bazel_skylib//lib:paths.bzl", _paths="paths")`

**Comment by manekinekko** [comment]
So, in theory, [this code](https://github.com/bazelbuild/rules_nodejs/blob/master/internal/node/node_repositories.bzl#L25) shouldn't trigger the deprecation warnings? Right?

Because we are getting these warning in https://github.com/angular/angular/issues/27603

**Comment by Capstan** [comment]
Correct. Unfortunately, you'll still get the warning if any starlark in your transitive closure of loads loads lib.bzl, including in the `WORKSPACE` file, IIUC.

**Comment by manekinekko** [comment]
Gotcha. Thanks for the feedback.

**Comment by laurentlb** [comment]
In my experience, the `print` warning was not useful. Users should deal with it when they upgrade (so it should be an error). Some people didn't, because it's a warning and maybe they didn't see it. As a result, anyone building their code gets a notification in the console. It's quite noisy and it encourages everyone to ignore the messages. For most people seeing the message, it's not actionable, because it's not their code.

So let's replace it with `fail` now.
